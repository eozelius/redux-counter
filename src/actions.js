// Action types
export const INCREMENT = 'INCREMENT'
export const DECREMENT = 'DECREMENT'

// Action Creators
export const incrementCounter = () => {
  return {
    type: INCREMENT
  }
}

export const decrementCounter = () => {
  return {
    type: DECREMENT
  }
}
