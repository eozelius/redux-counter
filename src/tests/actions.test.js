import configureStore from 'redux-mock-store'
import { INCREMENT, DECREMENT, incrementCounter, decrementCounter } from '../actions.js'

describe('redux Counter Actions', () => {
  const mockStore = configureStore();
  const store = mockStore();

  beforeEach(() => {
    store.clearActions()
  })

  it('produces an Increment action', () => {
    const expectedAction = [{
      type: INCREMENT
    }]
    store.dispatch(incrementCounter())
    expect(store.getActions()).toEqual(expectedAction)
  })

  it('produces a decrement action', () => {
    const expectedAction = [{
      type: DECREMENT
    }]
    store.dispatch(decrementCounter())
    expect(store.getActions()).toEqual(expectedAction)
  })
})
