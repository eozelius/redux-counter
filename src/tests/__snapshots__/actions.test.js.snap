// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`redux Counter Actions produces a decrement action 1`] = `
Array [
  Object {
    "type": "DECREMENT",
  },
]
`;

exports[`redux Counter Actions produces an Increment action 1`] = `
Array [
  Object {
    "type": "INCREMENT",
  },
]
`;
